libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -899,3 +951,430 @@ float AP_GPS::get_lag(uint8_t instance) const
         return drivers[instance]->get_lag();
     }
 }
+
+/*
+ calculate the weightings used to blend GPs location and velocity data
+*/
+void AP_GPS::calc_blend_weights(void)
+{
+    // zero the blend weights
+    memset(&_blend_weights, 0, sizeof(_blend_weights));
+
+    // determine how many GPS receivers are returning a fix and the instance of the first receiver that is
+    uint8_t first_instance = 0;
+    for (uint8_t i=0; i<GPS_MAX_RECEIVERS; i++) {
+        if (state[i].status > NO_FIX) {
+            num_instances = i+1;
+            if (first_instance == 0) {
+                first_instance = i;
```

Comment:
OK, I've removed the first_instance loop because the only case where it can be used is the case where it's value is zero.